File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.cpp
class Solution {
public:
    int maxProduct(int n) {
        if(n < 10) return n;
        int mx = INT_MIN;
        int smx = INT_MIN;

        while(n > 0){
            int rem = n % 10;
            if(rem > mx){
                smx = mx;
                mx = rem;
            } 
            else if(rem <= mx && rem >= smx) smx = rem;
            n /= 10;
        }

        return mx * smx;
    }
};

File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.py
class Solution:
    def maxProduct(self, n: int) -> int:
        if n < 10: 
            return n
        mx = 0
        smx = 0

        while n > 0:
            rem = n % 10
            if rem > mx:
                smx = mx
                mx = rem
            elif (rem <= mx and rem >= smx):
                smx = rem
            n //= 10

        return mx * smx